mllib-dal/src/main/native/oneapi/dal/HomogenTableImpl.cpp

```diff
@@ -191,12 +191,12 @@ JNIEXPORT jlong JNICALL Java_com_intel_oneapi_dal_table_HomogenTableImpl_fInit(
              break;
          }
          default: {
-             env->ReleaseFloatArrayElements(cData, fData, 0);
+             env->ReleasePrimitiveArrayCritical(cData, fData, 0);
```

Comment:
pls throw error for default. 
It's better to write error_message() function to print consistent error messages.